### PR TITLE
Tabs mechanism

### PR DESCRIPTION
**Pull request by ar4s**
Connected to #1450

### CODE REVIEW

File: src/ralph/admin/decorators.py [hunk 1]
```diff
@@ -6,8 +6,45 @@
...
+
+
+class register_extra_view(object):  # noqa
+    VIEWS = CHANGE, LIST = ('change', 'list')
+
+    def __init__(self, target_model=None, target_view=None):
+        if not all([target_model, target_view]):
+            raise ValueError('Please specify target_model and target_view.')
+        if Model not in target_model.__mro__:
+            raise ValueError('Model must be a inheritence from models.Model.')
```
Comment (mkurek) [inline src/ralph/admin/decorators.py]: inherited

File: src/ralph/admin/mixins.py [hunk 2]
```diff
@@ -21,26 +23,55 @@ class RalphAdminMixin(object):
...
     """Ralph admin mixin."""
 
-    extra_views = []
+    list_views = None
+    change_views = None
     change_list_template = 'ralph_admin/change_list.html'
     change_form_template = 'ralph_admin/change_form.html'
 
+    def get_list_views(self):
+        views = self.list_views and copy(self.list_views) or []
```
Comment (mkurek) [inline src/ralph/admin/mixins.py]: :hushed:

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+You must write your own class view and template. There are two possibilities to add extra view: by decorator and by class' attribute.
+
+### Your view
+
+It must be normal view (CBV) inherited from ``RalphListView`` or ``RalphDetailView`` (it depends on view purpose).
+
+#### RalphListView
+
+This class is deticated for listing view.
```
Comment (szok) [inline docs/development.md]: dedicated

File: src/ralph/tests/factories.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import factory
+
+from django.contrib.auth.models import User
```
Comment (szok) [inline src/ralph/tests/factories.py]: Please use: get_user_model()
django.contrib.auth.get_user_model

File: docs/development.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+Where:
+
+- ``app_label`` - is ``app_label`` extracted from ``model``,
+- ``model`` - lowercase name of ``model``,
+- ``name`` - name of view gets from view's instance.
+
+### Register view via class atrribiute
+
+**Note:** Use this method if you developing direct in Ralph.
```
Comment (mkurek) [inline docs/development.md]: directly?

File: src/ralph/admin/decorators.py [hunk 6]
```diff
@@ -6,8 +6,55 @@
...
+
+class WrongViewClassError(Exception):
+    pass
+
+
+class register_extra_view(object):  # noqa
+    VIEWS = CHANGE, LIST = ('change', 'list')
+
+    def __init__(self, target_model=None, target_view=None):
+        if not all([target_model, target_view]):
```
Comment (mkurek) [inline src/ralph/admin/decorators.py]: skip None default args and this validation

**Comment by vi4m** [comment]
This pull request is 21 days old! Could we make more-iterative approach?

**Comment by mkurek** [comment]
:ok:

**Comment by szok** [comment]
:ok:

**Pull request merged**